pkg/cmd/secrets_notes.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright © 2020 Doppler <support@doppler.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"bufio"
+	"os"
+	"strings"
+
+	"github.com/DopplerHQ/cli/pkg/configuration"
+	"github.com/DopplerHQ/cli/pkg/http"
+	"github.com/DopplerHQ/cli/pkg/printer"
+	"github.com/DopplerHQ/cli/pkg/utils"
+	"github.com/spf13/cobra"
+)
+
+var secretsNotesCmd = &cobra.Command{
+	Use:   "notes",
+	Short: "Manage secret notes",
+	Args:  cobra.NoArgs,
+}
+
+var secretsNotesSetCmd = &cobra.Command{
+	Use:   "set [secret] [note]",
+	Short: "Set a note on a secret. The secret must exist. Notes can be passed via arg or via stdin.",
+	Args:  cobra.RangeArgs(1, 2),
+	Run:   setSecretNote,
+}
+
+func setSecretNote(cmd *cobra.Command, args []string) {
+	jsonFlag := utils.OutputJSON
+	localConfig := configuration.LocalConfig(cmd)
+
+	utils.RequireValue("token", localConfig.Token.Value)
+
+	secret := args[0]
+	var note string
+	if len(args) > 1 {
+		note = args[1]
+	} else {
+		// read from stdin
+		hasData, e := utils.HasDataOnStdIn()
+		if e != nil {
+			utils.HandleError(e)
+		}
+
+		// note must be supplied
+		if !hasData {
+			utils.RequireValue("note", note)
+		}
+
+		var input []string
+		scanner := bufio.NewScanner(os.Stdin)
+		for {
+			if ok := scanner.Scan(); !ok {
+				if e := scanner.Err(); e != nil {
+					utils.HandleError(e, "Unable to read input from stdin")
+				}
+
+				break
+			}
+
+			s := scanner.Text()
+			input = append(input, s)
+		}
+
+		note = strings.Join(input, "\n")
```

Comment:
Do we have any other spots in the CLI that would benefit from taking stdin as input? If so, do you think it would be valuable to extract this out of `setSecretNote` to make it reusable by other commands?

Reply:
We currently support taking input via stdin in two other commands: `doppler secrets set` and `doppler configure set`. `doppler secrets set` would be difficult to refactor as it has a lot of custom logic, though `doppler configure set` is very similar to the new logic. I'll see if I can extract this logic for `doppler secrets notes set` and `doppler configure set`.

Reply:
Sweet! We can also do a follow-up refactor ticket for these if that sounds more reasonable!

Reply:
Just pushed up a commit for this